### ISSUE TITLE
Can i use yolov4 to detect colors?

### ISSUE THREAD

**Issue by adhilcodes** [AlexeyAB/darknet#6871]
Can i use yolov4 to detect colors?..  For example to detect traffic lights (the 3 colors)

**Comment by stephanecharette** [comment]
Yes.

Make sure you turn down or completely [turn off data augmentation related to hue](https://www.ccoderun.ca/darkmark/DataAugmentationColour.html), etc.

Then you make 3 classes, such as "red light", "green light", "amber light".

**Comment by adhilcodes** [comment]
@stephanecharette If I have turn off those 3 data augmentations (exposure, saturation and hue) does it the accuracy of the yolov4 model .

**Comment by shubham-shahh** [comment]
In your case, playing with hue values will actually make the model worse.

**Comment by shubham-shahh** [comment]
I would go with OpenCV, if all you want to do is, color detection.

**Comment by adhilcodes** [comment]
> I would go with OpenCV, if all you want to do is, color detection.

@shubham-shahh But But it doesn't give the speed of yolov4 algorithm

**Comment by shubham-shahh** [comment]
OpenCV with GPU might perform better. I mean using yolo for color detection is overkill.

**Comment by stephanecharette** [comment]
I disagree.  I think Darknet/YOLO is a great way to find and identify the different coloured traffic lights, and I would definitely prefer it over plain OpenCV.  This is exactly what Darknet/YOLO is for.  Like I wrote above, the only thing you'll need to do is remember to turn down or completely turn off the colour-based data augmentation options.

Depending on the range of images you are working with, the lighting conditions outside, etc..., only you can determine if it needs to be completely turned off or simply set to a low value.  Experiment and see.

**Comment by adhilcodes** [comment]
> I disagree.  I think Darknet/YOLO is a great way to find and identify the different coloured traffic lights, and I would definitely prefer it over plain OpenCV.  This is exactly what Darknet/YOLO is for.  Like I wrote above, the only thing you'll need to do is remember to turn down or completely turn off the colour-based data augmentation options.
> 
> Depending on the range of images you are working with, the lighting conditions outside, etc..., only you can determine if it needs to be completely turned off or simply set to a low value.  Experiment and see.

So If I switch of **exposure, saturation** and **Hue** can I can get a better accuracy

**Comment by stephanecharette** [comment]
> @stephanecharette So If I switch of **exposure, saturation** and **Hue** can I can get a better accuracy in YOLO V4

No, this has nothing to do with "better accuracy".  The problem with the colour values for data augmentation is if you mark a traffic light in your image as "green light", then a large value of "hue" will change all greens to blue, or change all greens to pink, or change all greens to red.....and then the network won't train correctly because a "red" light has been annotated the same way as a "green" light.

So if your classes are based on colour, then you need to turn down or completely turn off data augmentation based on hue.  See the link I posted in my very first comment at the top of this issue to have an idea of what different values of "hue" will do to your images during training.

**Comment by adhilcodes** [comment]
@stephanecharette  I have tried with different values for hue . I got better mAP at Hue=0.4, saturation=1.5& exposure=1.5

**Comment by adhilcodes** [comment]
@stephanecharette  I have tried with different values for hue . I got better mAP at Hue=0.4, saturation=1.5& exposure=1.5
